### PR TITLE
feat: new Twitter follow/like/retweet requirements

### PR DESCRIPTION
**Pull request by BrickheadJohnny**
Added 3 new Twitter requirement types:
- `TWITTER_FOLLOW_V2`
- `TWITTER_LIKE_V2`
- `TWITTER_RETWEET_V2`

The API rate limits are quite low, so we decided to use [Web Intents](https://developer.twitter.com/en/docs/twitter-for-websites/web-intents/overview). The main logic for these new types is inside the `TwitterIntent` component.

### CODE REVIEW

File: src/requirements/Twitter/components/TwitterIntent.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    onSuccess: () => {
+      mutateAccess()
+      setHasClicked(false)
+    },
+  })
+
+  const [hasClicked, setHasClicked] = useState(false)
+  // Calling the callback endpoint only on refocus
+  useSWR(
+    hasClicked ? ["twitterRequirement", requirementId] : null,
```
Comment (cs-balazs) [inline src/requirements/Twitter/components/TwitterIntent.tsx:82]: Not sure, but maybe include the `userId` here as well? I guess it only matters if the user switches to a different account, and clicks again(?)

File: src/requirements/Twitter/components/TwitterIntent.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      revalidateOnMount: false,
+      revalidateIfStale: false,
+      revalidateOnReconnect: false,
+      revalidateOnFocus: true,
+      refreshInterval: 0,
+    }
+  )
+
+  const onClick = () => {
+    setHasClicked(true)
```
Comment (cs-balazs) [inline src/requirements/Twitter/components/TwitterIntent.tsx:101]: Do we trigger a `completeAction` call here in all cases? I mean, even if the user has access. If so, is it intended?

**Review by cs-balazs — APPROVED** [review]
Great changes!
Approving, as my comments are just minor questions  🚀

File: src/requirements/Twitter/components/TwitterIntent.tsx (hunk 2 above)
Comment (BrickheadJohnny) [reply src/requirements/Twitter/components/TwitterIntent.tsx:101]: yup, but that's why I check `hasAccess` in line 84 ;)

File: src/requirements/Twitter/components/TwitterIntent.tsx (hunk 1 above)
Comment (BrickheadJohnny) [reply src/requirements/Twitter/components/TwitterIntent.tsx:82]: Oh, good point, will make sure to add it here!

File: src/requirements/Twitter/components/TwitterIntent.tsx (hunk 2 above)
Comment (cs-balazs) [reply src/requirements/Twitter/components/TwitterIntent.tsx:101]: That makes sense, thank you!

**Pull request merged**